### DISCUSSION TITLE
How are destination rule credentials received by the data plane?

### DISCUSSION THREAD

**Discussion by wtzhang23** [istio/istio#52048] [Q&A] [answered]
Specifically, the [ClientTlsSettings](https://istio.io/latest/docs/reference/config/networking/destination-rule/#ClientTLSSettings)'s `credentialName` field. Is the secret fetched by proxy agent (the data plane) or Istiod? A cursory glance at the code base suggests it is retrieved by Istiod and shared to the data plane over SDS. This would imply that Istiod would need an RBAC policy to read all secrets in a cluster (already given in Helm [chart](https://github.com/istio/istio/blob/0a1dadde9dd9d7480574b0f43824e16120d86750/manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml#L129-L131))? And furthermore, the service account of the service that receives that credential doesn't necessarily need an RBAC binding for that credential? Not necessarily a problem if so; just curious.

**Comment by howardjohn** [comment, marked as answer]
It is fetched by Istiod, which does already have RBAC permissions to do so as you mentioned.

The reason the individual workload needs the permission is to prove to Istiod it is authorized to access it. Specfically, we do not want arbitrary pods requesting arbitrary Secrets from Istiod
